dlang/psi-impl/src/main/java/io/github/intellij/dlanguage/psi/impl/named/DLanguageFunctionDeclarationImpl.java

```diff
@@ -114,8 +115,10 @@ public boolean processDeclarations(@NotNull final PsiScopeProcessor processor, @
     @Override
     public @NotNull DType getReturnDType() {
         var basicType = getBasicType();
-        if (basicType != null)
-            return basicType.getDType();
+        if (basicType != null) {
+            var type = basicType.getDType();
+            return DTypesUtilKt.getDtypeWithSuffixes(type, getTypeSuffixes());
```

Comment:
The fix, to be able to determine array type (or pointer, ...)  for function return types